alibi_detect/utils/tensorflow/prediction.py

```diff
@@ -1,14 +1,40 @@
+import inspect
 from functools import partial
-from typing import Callable, Type, Union
+from typing import Any, Callable, Dict, Type, Union
 
 import numpy as np
 import tensorflow as tf
 from alibi_detect.utils.prediction import tokenize_transformer
 
 
-def predict_batch(x: Union[list, np.ndarray, tf.Tensor], model: Union[Callable, tf.keras.Model],
-                  batch_size: int = int(1e10), preprocess_fn: Callable = None,
-                  dtype: Union[Type[np.generic], tf.DType] = np.float32) -> Union[np.ndarray, tf.Tensor, tuple]:
+def get_named_arg(model: tf.keras.Model, x: Any) -> Dict[str, Any]:
+    """ Extract argument names from the model call function
+    because keras3 does not accept other types of input
+    as a positional argument.
+
+    Parameters
+    ----------
+    model
+        Model to extract argument names from.
+    x
+        Input data.
+
+    Returns
+    -------
+    Dictionary with named arguments.
+    """
+    signature = inspect.signature(model.call)
+    arg_names = [param.name for param in signature.parameters.values()]
+    return {arg_names[0]: x}
```

Comment:
question: what are the guarantees that we are always interested in the first arg?

Reply:
This construct it is used where the model is called with a single argument.